spec/bolt_server/transport_app_spec.rb

```diff
@@ -222,26 +245,74 @@ def file_data(file)
         expect(last_response).to be_ok
         expect(last_response.status).to eq(200)
       end
+
+      it 'expects either a single target or a set of targets, but not both' do
+        single_target = {
+          hostname: target[:host],
+          user: target[:user],
+          password: target[:password],
+          port: target[:port]
+        }
+        body = { target: single_target }
+
+        post(path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json')
+        expect(last_response.status).to eq(200)
+
+        body = { targets: [single_target] }
+
+        post(path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json')
+        expect(last_response.status).to eq(200)
+
+        body = { target: single_target, targets: single_target }
+
+        post(path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json')
```

Comment:
With regards to https://github.com/puppetlabs/bolt/pull/1221/commits/58adf03e13caa3f69f5c864faa899d757652be40 , I think its pretty bad that we regressed on that so bad. (for example with a typo in the schema you would get)
```
cas@cas-ThinkPad-T460p:~/working_dir/bation/json" -d @request.json --cacert $BOLT_CACERT --cert $BOLT_CERT --key $BOLT_KEY $BOLT_ROOT/winrm/run_task
{"kind":"boltserver/server-error","msg":"500: Unknown error: undefined method `status_hash' for 400:Integer","details":{}}
```
I see that we are expecting a status 400 (where it would have been 500 before), maybe we should also do some regex matching on the body of the response `expect(last_response.body).to match(/type object matched more than one of the required schemas/)` to ensure we are surfacing the message from the schema validation. Or possibly we need to just test schema validation error handling somewhere more generally.

Reply:
I've added one test each for SSH and WinRM as you describe, here:
https://github.com/puppetlabs/bolt/pull/1221/files#diff-3afac83384bfe4aaafb92f6c7c6d5225R373
Anything else you'd like to see in this PR?

Reply:
Looks great. Thanks!